students/Gregg/lesson3/mailroom.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+"""Use current knowledge to set up an interactive program"""
+
+donations = {
+    "William Gates, III": [3, 5, 7],
+    "Mark Zuckerberg": [4.50, 8, 2],
+    "Jeff Bezos": [7.77],
+    "Paul Allen": [3.6, 4.5],
+    "bob": [.01]
+}
+return_to_primary_prompt = [False]
+
+# This doesn't seem like the cleanest way to do this, but I coultn't
+# come up with anything better without turning it into a full blow stat machine
+# that has a centralized handler thingy for every possible transition
+# That can then return to the start state if necessary
+# It seems like it works ok for this limited example
+
+
+def quitable_input(text_prompt):
+    text_in = input(f"{text_prompt}\nInput: ")
+    if text_in == 'q':
+        return_to_primary_prompt[0] = True
+        return False
+    else:
+        return text_in
+
+
+def primary_prompt():
+    """Text cue to help user select an action"""
+    prompt_string = (
+        "\nWould you like to Send a Thank You (enter TY), "
+        "Create a Report (enter CR) or quit (enter Q).\n"
+        "At any point enter q to return to this prompt\n"
+        "Input: "
+    )
+    return prompt_string
+
+
+def thank_you_prompt():
+    """Text cue to help user send a thankyou"""
+    prompt_string = (
+        "Enter a full name to send them a thank you\n"
+        "Enter 'list' to see current donors"
+    )
+    return prompt_string
+
+
+def donation_amount_prompt():
+    """Text cue to help user enter a donation amount"""
+    prompt_string = "How much did this person donate?"
+    return prompt_string
+
+
+def handle_primary_prompt(action):
+    """Handle the possible user responses to the primary prompt"""
+    map_inputs = {
+        "TY": handle_thankyou_prompt,
+        "CR": create_donation_report
+    }
+    action_handled = False
+    try:
+        action_handled = map_inputs[action]()
+    except KeyError as E:
+        print(f"{action} That isn't a valid input")
+    return action_handled
+
+
+def handle_thankyou_prompt():
+    """Handle the possible user responses to the thankyou prompt"""
+    text_in = quitable_input(thank_you_prompt())
+    if return_to_primary_prompt[0]:
+        return False
```

Comment:
your `quitable_input` function already returns an indicator to quit - I think you should get rid of the global `return_to_primary_prompt` variable